kasa/smart/smartdevice.py

```diff
@@ -158,7 +164,13 @@ async def _initialize_modules(self):
         """Initialize modules based on component negotiation response."""
         from .smartmodule import SmartModule
 
+        # Initialise the CloudModule
+        cloudModule = CloudModule(self, CloudModule.REQUIRED_COMPONENT)  # noqa: F405
+        self.modules[cloudModule.name] = cloudModule
+
         for mod in SmartModule.REGISTERED_MODULES.values():
+            if mod == CloudModule:  # noqa: F405
+                continue
```

Comment:
Ohh, now I see the problem.. Uhh, but I think we don't really need this type of special handling. The `is_cloud_connected` already checks for the existence of the module, so it'll get initialized if it's available, and return False otherwise. Or am I missing something? :-)

Reply:
I was trying to avoid the firmware module doing the `get_latest_fw` call if the device doesn't have the `cloud_connect` component but as that's probably not going to happen have removed this.